src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/SearchTree.cs

```diff
@@ -94,10 +96,16 @@ public string SearchForPattern(string term)
         /// <exception cref="ArgumentException"> if <paramref name="term"/> is null, empty, or whitespace only.</exception>
         public IReadOnlyList<string> GetConfiguredPackageSources(string term)
         {
-            return SearchNodeByTerm(term)?.PackageSources;
+            SearchNode? searchNodeResult = SearchNodeByTerm(term);
```

Comment:
I see this makes my above comment invalid. 

I think if we're going to have a refactoring like this, it makes to do it in a separate PR. 

Furthermore, this *is* a binary breaking change. The behavior of the method is changing, even if undeclared. 

Is there a place where we differentiate between empty and null? Is that ever relevant or do they largely mean the same thing?

Reply:
> I think if we're going to have a refactoring like this, it makes to do it in a separate PR.

Can do

> Is there a place where we differentiate between empty and null? Is that ever relevant or do they largely mean the same thing?

"Same thing", but I'm trying to consolidate these APIs to return empty to avoid unnecessary null checking plus empty checking by callers. 
I should be using Empty checks everywhere.